torch/csrc/utils/tensor_new.cpp

```diff
@@ -1710,7 +1710,9 @@ Tensor asarray(
       if (wrong_device || wrong_dtype) {
         tensor = tensor.to(
             device.value_or(tensor.device()),
-            dtype.value_or(tensor.scalar_type()));
+            dtype.value_or(tensor.scalar_type()),
+            /*non_blocking=*/false,
+            /*copy=*/force_copy);
```

Comment:
Minor change: now, I think you don't really need this inner-most if-else, since it's using the `Tensor.to` method for figuring that out. Won't block because of this, though.

Reply:
Thank you.:D